README.md

```diff
@@ -367,3 +367,31 @@ Promise.resolve("Hello!")
 ```
 
 If you don't `catch` a rejected promise, it will throw an error and crash (the same as if you didn't catch an exception in a try/catch statement).
+
+# Persisting the state
+
+Let's make sure that changes in our list are persisted between apps restarts. To do that, we will be using [`AsyncStorage`](https://facebook.github.io/react-native/docs/asyncstorage.html).
+
+To retrieve something stored, we can use `getItem`: it accepts a key and it returns a promise, the result will be either the previously stored data, or `null` if there isn't any.
+
+To retrieve the state when the application is initialized, we can use one of the lifecycle callbacks of a component (they're present only in class components), [`componentDidMount`](https://reactjs.org/docs/react-component.html#componentdidmount) and set the retrieved state inside the promise callback.
+
+```javascript
+componentDidMount() {
+  AsyncStorage.get('todos')
+    .then(todos => this.setState({todos: todos}));
+}
+```
+
+To store something, we can use `setItem`: it accepts a key and the value that we want to store, it returns a promise that will be resolved when the data is stored (in this case, when the promise is resolved, we won't receive any value).
+
+There isn't a standard place for us to store the state, there is [`componentWillUnmount`](https://reactjs.org/docs/react-component.html#componentwillunmount) but the JS machine won't wait for us to save the state. One possible option is to save the state every time we update it.
```

Comment:
I do not fully get this, we can use `setItem` to store value and key but we can't set state? Is that correct?

Reply:
Hm basically there are 2 things:

1. You're updating the state of the **component**, this is done by `setState`
2. You're trying to persist the state so that it can be restored at app startup, one solution (not the only one) is to use `AsyncState` and save a value.

The two things aren't related (you can save any value with `AsyncStorage`). The problem with `componentWillUnmount` is that it won't wait for you to save to disk, so in my experience what tends to happen is that the operation never finishes and, when you try to restore that data, it isn't there

Reply:
I see, I think then that it should be a little more clear on the docs that you are doing two things here. Because I was trying to figure out how they related to each other instead of reading it as two steps.

Reply:
I see what happened, my brainfart:

> There isn't a standard place for us to store the state

this should be

> There isn't a standard lifecycle method we can use to persist our state

Reply:
Yeah that sounds much better 😄

Reply:
840d40c